src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -206,9 +207,95 @@ ss::future<server::reply_t> get_mode(server::request_t rq, server::reply_t rp) {
     parse_accept_header(rq, rp);
     rq.req.reset();
 
-    rp.rep->write_body("json", ss::sstring{R"({
-  "mode": "READWRITE"
-})"});
+    // Ensure we see latest writes
+    co_await rq.service().writer().read_sync();
+
+    auto res = co_await rq.service().schema_store().get_mode();
+
+    auto json_rslt = ppj::rjson_serialize(mode_req_rep{.mode = res});
+    rp.rep->write_body("json", json_rslt);
+    co_return rp;
+}
+
+ss::future<server::reply_t> put_mode(server::request_t rq, server::reply_t rp) {
+    parse_content_type_header(rq);
+    parse_accept_header(rq, rp);
+    auto frc = parse::query_param<std::optional<force>>(*rq.req, "force")
+                 .value_or(force::no);
+    auto res = ppj::rjson_parse(rq.req->content.data(), mode_handler<>{});
+    rq.req.reset();
+
```

Comment:
> I'm still trying to wrap my head around how these sequence markers work, but is the reason why we don't need a `co_await rq.service().writer().read_sync();` here because this write is global and not subject specific?

The `seq_writer::write_*` calls all call `read_sync()` internally.